packages/server-core/src/media/static-resource-search/static-resource-search.class.ts

```diff
@@ -0,0 +1,253 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2025 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import { BadRequest, NotImplemented } from '@feathersjs/errors'
+import { Id, NullableId, Params, ServiceMethods } from '@feathersjs/feathers'
+import {
+  DEFAULT_SEARCH_PARAMS,
+  StaticResourceSearchQueryType,
+  StaticResourceSearchResponseType,
+  StaticResourceSearchResultType
+} from '@ir-engine/common/src/schemas/media/static-resource-search.schema'
+import { staticResourceVectorPath } from '@ir-engine/common/src/schemas/media/static-resource-vector.schema'
+import { staticResourcePath } from '@ir-engine/common/src/schemas/media/static-resource.schema'
+import { Application } from '../../../declarations'
+import { getStorageProvider } from '../storageprovider/storageprovider'
+
+export interface StaticResourceSearchParams extends Params {
+  query?: StaticResourceSearchQueryType
+}
+
+export class StaticResourceSearchService implements Partial<ServiceMethods<StaticResourceSearchResultType>> {
+  app: Application
+
+  constructor(app: Application) {
+    this.app = app
+  }
+
+  /**
+   * Find static resources using semantic search
+   */
+  async find(params?: StaticResourceSearchParams): Promise<StaticResourceSearchResponseType> {
```

Comment:
This is throwing a type error as the feathers app is expecting a return type of Promise<StaticResourceSearchResultType>